### ISSUE TITLE
[Build-in Function] "values()" dose not support the map element value is TypeList

### ISSUE THREAD

**Issue by LanceNero** [hashicorp/terraform#15395]
Hi there,

### Terraform Version
Terraform v0.9.6

### Terraform Configuration Files
```hcl
variable "test_map" {
  type = "map"
  description = "describe your variable"
  default = {
    key1 = ["val1","val3"]
    key2 = ["val2"]
  }
}


resource "null_resource" "test" {

   provisioner "local-exec" {
       command = "echo ${join(",",values(var.test_map))}"
   }
}

```

### Expected Behavior
I hope "values()" function  can return all the values of this map
like this: [ "val1","val3","val2"]

But it returned an error

### Debug Output
> 1 error(s) occurred:
> * null_resource.test: 1 error(s) occurred:
> * null_resource.test: values: values(): "key1" has element with bad type TypeList in:
> echo ${join(",",values(var.test_map))}

### Steps to Reproduce
 `terraform apply`

I did not find any other way to achieve  this scenes
now the element value of map has been supported TypeList 
So, can the related functions support these types? 

If there are other ways to achieve that and I did not notice, please also remind  me
thank u so much ~

**Comment by AlexRudd** [comment]
Having a problem with this too. Here's my use case:

tfvars:
```
admin_users = {
  "account1" = ["user1", "user2", "user3"]
  "account2" = ["user1", "user2", "user3"]
  "account3" = ["user1"]
}

read_only_users = {
  "account1" = ["user4", "user5"]
  "account2" = ["user4", "user5"]
  "account3" = []
}
```

tf:
```
locals {
  all_users = "${distinct(
    concat(
      flatten(values(var.admin_users)),
      flatten(values(var.read_only_users))
    )
  )}"
}
```

apply output:
```
* local.all_users: local.all_users: values: values(): "account1" has element with bad type TypeList in:

${distinct(
    concat(
      flatten(values(var.admin_users)),
      flatten(values(var.read_only_users))
    )
  )}
```

**Comment by apparentlymart** [comment]
Hi all!  Sorry for the delayed response.

In v0.12.0-alpha1 I've verified that the `values` function now supports keys of any type. Here is an example using `terraform console` just for simplicity:

```
> values({"a":[], "b": ["a"]})
[
  [],
  [
    "a",
  ],
]
```

In reproducing this I discovered a new issue #19204 which we'll need to fix before v0.12.0 final, but since  the original problem is now fixed in master and ready to be included in the final release I'm going to close this. Thanks for reporting this, and thanks for the patience while we laid the groundwork to fix it.
